File: src/main/java/com/isaac/games/CardGraphic.java
package com.isaac.games;

import javafx.scene.image.ImageView;

public class CardGraphic extends ImageView {

    private Card card;

    public CardGraphic(Card card) {
        super("images/" + sanitize(card.toString()) + ".bmp");
        this.card = card;
    }

    public Card card() {
        return card;
    }

    @Override
    public String toString() {
        return card.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof CardGraphic)) {
            return false;
        }
        CardGraphic that = (CardGraphic) o;
        return this.card.equals(that.card);
    }

    private static String sanitize(String input) {
        int x = input.indexOf('$');
        return (x == -1) ? input : input.substring(0, x);
    }
}

File: src/main/java/com/isaac/games/CardGraphic.kt
package com.isaac.games

import javafx.scene.image.ImageView

class CardGraphic(private val card: Card) : ImageView("images/" + sanitize(card.toString()) + ".bmp") {

    fun card(): Card {
        return card
    }

    override fun toString(): String {
        return card.toString()
    }

    override fun equals(o: Any?): Boolean {
        if (o !is CardGraphic) {
            return false
        }
        val that = o as CardGraphic?
        return this.card == that!!.card
    }

    companion object {
        private fun sanitize(input: String): String {
            val x = input.indexOf('$')
            return if (x == -1) input else input.substring(0, x)
        }
    }
}